assignment+4/InvertedIndex.java

```diff
@@ -1,50 +1,40 @@
-package assignment4;
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
 
 // list of terms, pointing to the respective record
-public class InvertedIndex {
+public class InvertedIndex<T> {
 
-  public Entry[] entries;
+  private static final String[] STOP_WORDS = { "a", "la", "le", "de", "des", "the", "aux", "du", "of", "sur" };
+  public ArrayList<Entry> entries;
 
   private class Entry {
     private String term;
-    private Record[] documents;
+    private ArrayList<T> documents;
 
     // constructor for Entry object
-    private Entry(String term, Record[] documents) {
+    private Entry(String term, T document) {
       this.term = term;
-      this.documents = documents;
+      this.documents = new ArrayList<T>();
+      this.documents.add(document);
     }
-
-    private void printEntry() {
-      System.out.printf("Term: %s", term);
-      System.out.println();
-      System.out.println("Documents:");
-      for (Record d : documents) {
-        d.printRecord();
-      }
+    @Override
```

Comment:
Is this necessary put whenever you do a toString() function?

Reply:
every object in java inherits the toString() function from the Object superclass (parent of all objects in Java).

if you want to override what happens when a call on your object, like `System.out.println(object)` behaves, then you need to override the toString() method.

i thought it would be cleaner to have a line of code like:
`System.out.println(record)'

than to have something like

`System.out.println(record.cgndbid + record.latitude)` etc etc